p1-staking-pool-dyn/src/lib.rs

```diff
@@ -174,7 +163,8 @@ impl Contract {
     #[payable]
     pub fn close(&mut self) -> Promise {
         assert_one_yocto();
-        let (aid, mut vault) = self.get_vault();
+        let aid = env::predecessor_account_id();
+        let mut vault = self.get_vault_or_default(&aid);
```

Comment:
as above, default doesn't make sense for this function.

Reply:
if there's no vault, it will fail because to_transfer == 0